clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java

```diff
@@ -3318,6 +3351,39 @@ public void fetchVariables() {
         fetchVariables(null);
     }
 
+    /**
+     * Fetches In Apps from server.
+     *
+     * @param callback Callback instance {@link FetchInAppsCallback} to be invoked when fetching is done.
+     */
+    public void fetchInApps(FetchInAppsCallback callback) {
+        if (coreState.getConfig().isAnalyticsOnly()) {
+            return;
+        }
+        Logger.v(Constants.LOG_TAG_INAPP + " Fetching In Apps...");
+
+        if (callback != null) {
+            coreState.getCallbackManager().setFetchInAppsCallback(callback);
+        }
+
+        JSONObject event = getFetchRequestAsJson(Constants.FETCH_TYPE_IN_APPS);
+        coreState.getAnalyticsManager().sendFetchEvent(event);
+    }
+
+    @NonNull
+    private JSONObject getFetchRequestAsJson(int fetchType) {
+        JSONObject event = new JSONObject();
+        JSONObject notif = new JSONObject();
+        try {
+            notif.put(Constants.KEY_T, fetchType);
+            event.put(Constants.KEY_EVT_NAME, Constants.WZRK_FETCH);
+            event.put(Constants.KEY_EVT_DATA, notif);
+        } catch (JSONException e) {
+            e.printStackTrace();
```

Comment:
use logger and lets not print the stacktrace like this.

Reply:
added changes in https://github.com/CleverTap/clevertap-android-sdk/commit/21b72bddf52fd017776007fefa0bc6baa4deac20